plugins/pdi-pur-plugin/src/org/pentaho/di/ui/repository/pur/repositoryexplorer/controller/TrashBrowseController.java

```diff
@@ -384,6 +388,27 @@ public void undelete() {
           if ( dir != null ) {
             dirMap.get( dir.getObjectId() ).refresh();
           }
+          // if transformation or directory with transformations call extension to restore data services references.
+          if ( RepositoryObjectType.TRANSFORMATION.name().equals( uiObj.getType() ) ) {
+            TransMeta transMeta = repository.loadTransformation( uiObj.getId(), null );
+            ExtensionPointHandler
+                .callExtensionPoint( LogChannel.GENERAL, KettleExtensionPoint.TransAfterOpen.id, transMeta );
+            transMeta.clearChanged();
+          } else if ( !RepositoryObjectType.JOB.name().equals( uiObj.getType() ) ) {
+            // if not a transformation and not a job then is a Directory
+            RepositoryDirectoryInterface
+                actualDir =
```

Comment:
If uiObj is not a directory, or is not found, .findDirectory() will return null.
We should check for null and handle it.